src/jit/importer.cpp

```diff
@@ -1951,6 +1951,12 @@ bool                Compiler::impSpillStackEntry(unsigned   level,
     /* Assign the spilled entry to the temp */
     impAssignTempGen(tnum, tree, verCurrentState.esStack[level].seTypeInfo.GetClassHandle(), level);
 
+    // If the tree contains var address on stack, propagate such flag to a local which is spilled.
+    if (tree->IsVarAddr())
+    {
+        lvaTable[tnum].lvIsVarAddrOnStack = true;
+    }
```

Comment:
Another possible il pattern

lcarg.0   // load the arg 0 of type byref say
conv.r8